tests/Subscriber/FrontendSubscriberTest.php

```diff
@@ -34,10 +28,45 @@ class FrontendSubscriberTest extends TestCase
     use SalesChannelHelper;
     use CategoryHelper;
 
+    public function headerPageletLoadedEventProvider(): array
+    {
+        return [
+            'Search Request' => [
+                'requestParams' => [
+                    ['search' => 't-shirt'],
+                    [],
+                    [],
+                    [],
+                    [],
+                    ['REQUEST_URI' => 'https://example.com/search']
+                ],
+                'expectedPageInformation' => [
+                    'isSearchPage' => true,
+                    'isNavigationPage' => false
+                ]
+            ],
```

Comment:
Please add test examples for all cases. (e.g. request url includes "/search", but search parameter is not given).

Would be great, if you could split the test cases into a new test function. In this function, you could just skip the check for `PageInformation`.